homeassistant/components/ecobee/climate.py

```diff
@@ -206,16 +209,20 @@ def __init__(self, data, thermostat_index, hold_temp):
         self._fan_modes = [FAN_AUTO, FAN_ON]
         self.update_without_throttle = False
 
-    def update(self):
+    async def async_update(self):
         """Get the latest state from the thermostat."""
         if self.update_without_throttle:
-            self.data.update(no_throttle=True)
+            await self.data.update(no_throttle=True)
             self.update_without_throttle = False
         else:
-            self.data.update()
-
+            await self.data.update()
         self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
 
+    async def option_update_listener(self, hass, entry):
+        """Update relevant options in this entity."""
+        _LOGGER.debug("Updating options in ecobee thermostat {}".format(self.name))
+        self.hold_temp = entry.options[CONF_HOLD_TEMP]
```

Comment:
Where are we using `self.hold_temp`?

Reply:
You know I was just looking at that very thing and it turns out... it's not being used??

Looking at the previous code, there is no mention of self.hold_temp anywhere in the entity class, and that parameter (hold_temp) is only passed to the climate platform. I suspect it should find it's way into the entity method `hold_preference` - but it's even unclear how given the comments on that method.

My guess (I haven't looked at the commit history) is that hold_temp was once a part of hold_preference, but then hold_preference was changed and hold_temp just hung around.

This makes things easier - drop that config parameter, drop the options flow.

Reply:
I just pushed a batch of commits that removes all references to hold_temp and takes out the options flow code (since there are no other options to set for the ecobee integration at this time).